### PR TITLE
Create and delete new workspace test

### PR DESCRIPTION
**Pull request by nataliafz**
https://globalfishingwatch.atlassian.net/browse/MAP-1220

https://github.com/GlobalFishingWatch/frontend/assets/90259114/5bc3199b-f032-49eb-95eb-3d50270b18eb

### CODE REVIEW

File: apps/fishing-map/features/workspace/shared/NewWorkspaceModal.tsx [hunk 1]
```diff
@@ -167,6 +167,7 @@ function NewWorkspaceModal({
         value={name}
         label={t('common.name', 'Name')}
         className={styles.input}
+        testId="create-workspace-inpunt"
```
Comment (rdgfuentes) [inline apps/fishing-map/features/workspace/shared/NewWorkspaceModal.tsx:170]: There's a typo , it should be `create-workspace-input`

File: apps/fishing-map-e2e/src/e2e/sidebar/save-workspace.cy.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  it('Should create a workspace', () => {
+    cy.getBySel('save-workspace-button').click()
+    cy.getBySel('create-workspace-inpunt').clear()
+    cy.getBySel('create-workspace-inpunt').type(WORKSPACE_NAME)
+    cy.getBySel('create-workspace-button').click()
+    getWorkspace().should('contain', WORKSPACE_NAME)
+  })
+
+  it('Should delete a workspace', () => {
```
Comment (rdgfuentes) [inline apps/fishing-map-e2e/src/e2e/sidebar/save-workspace.cy.ts:28]: Tests should be independent and not depend from others. This one is depending on the preious test that creates the workspace. I'd merge both tests into one

**Review by rdgfuentes — CHANGES_REQUESTED** [review]
@nataliafz Nice and clean actions for tests make them super legible :+1: 
Thanks for using `testId` :muscle:
Check my comments above asking for some changes.

File: apps/fishing-map-e2e/src/e2e/sidebar/save-workspace.cy.ts (hunk 2 above)
Comment (fertandil87) [reply apps/fishing-map-e2e/src/e2e/sidebar/save-workspace.cy.ts:28]: and as we talked in the meeting the workspaces should be deleted before the test

File: apps/fishing-map-e2e/src/e2e/sidebar/save-workspace.cy.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  beforeEach(() => {
+    disablePopups()
+    cy.visit('/')
+    waitForSidebarLoaded()
+    waitForMapLoadTiles()
+  })
+
+  it('Should create a workspace', () => {
+    cy.getBySel('save-workspace-button').click()
+    cy.getBySel('create-workspace-inpunt').clear()
```
Comment (fertandil87) [inline apps/fishing-map-e2e/src/e2e/sidebar/save-workspace.cy.ts:22]: inpunt typo

File: apps/fishing-map-e2e/src/e2e/sidebar/save-workspace.cy.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    cy.getBySel('create-workspace-button').click()
+    getWorkspace().should('contain', WORKSPACE_NAME)
+  })
+
+  it('Should delete a workspace', () => {
+    cy.get('a[href*="/user"]').contains(Cypress.env('apiUserInitials')).click()
+    cy.getBySel('user-workspace').click()
+    cy.get('a[href*="/fishing-activity"]').contains(WORKSPACE_NAME)
+    cy.getBySel('remove-workspace-button').click()
+    cy.getBySel('user-workspaces').should('contain', 'Your workspaces will appear here')
```
Comment (fertandil87) [inline apps/fishing-map-e2e/src/e2e/sidebar/save-workspace.cy.ts:33]: is there another way to verify? in my case tests are run in spanish and this fails

Comment (fertandil87) [reply apps/fishing-map-e2e/src/e2e/sidebar/save-workspace.cy.ts:33]: also a translations can change any time

File: apps/fishing-map-e2e/src/support/app.po.ts [hunk 5]
```diff
@@ -2,6 +2,8 @@ import { API_URL_4WINGS_TILES } from '../constants/urls'
 
 export const getGreeting = () => cy.get('h1')
 
+export const getWorkspace = () => cy.get('h2')
```
Comment (fertandil87) [inline apps/fishing-map-e2e/src/support/app.po.ts:5]: this should use a test attribute

**Review by fertandil87 — CHANGES_REQUESTED** [review]

File: apps/fishing-map-e2e/cloudbuild.yaml [hunk 6]
```diff
@@ -60,6 +60,7 @@ steps:
     waitFor: ['restore_cypress_cache']
     entrypoint: 'bash'
     env:
+      - 'CYPRESS_publicApiGateway=$_NEXT_PUBLIC_API_GATEWAY'
```
Comment (j8seangel) [inline apps/fishing-map-e2e/cloudbuild.yaml:63]: As we are using the `NEXT_PUBLIC_API_GATEWAY` env variable is [this line](https://github.com/GlobalFishingWatch/frontend/pull/2170/files#diff-308681de6a66133e880aabfcbdc55961cdc5ca61b838a4b36d90bba82e523c09R10) used?

**Review by j8seangel — COMMENTED** [review]
Am I missing something? because I got this error running it locally
image

This was the screenshot, in case it is useful:
image

**Comment by j8seangel** [comment]
> Am I missing something? because I got this error running it locally image


@fertandil87 this seems to fix it, can you review it?
https://github.com/GlobalFishingWatch/frontend/commit/23cd70bea3654064137df68210b15798453fc5a3

**Review by j8seangel — APPROVED** [review]
LGTM

**Review by fertandil87 — APPROVED** [review]

**Pull request merged**